app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -19,6 +19,14 @@ def plan(capsule_content, options = {})
 
           fail _("Action not allowed for the default capsule.") if capsule_content.default_capsule?
 
+          need_importer_update = repos_needing_importer_updates(capsule_content, environment, content_view)
+          need_distributor_update = repos_needing_distributor_updates(capsule_content, environment, content_view)
+          need_refresh = (need_distributor_update && need_importer_update).uniq
```

Comment:
Pushing these three lines into a function that returns repositories needing update could help readability and hide away those details to keep the plan function succinct.